### PR TITLE
Fix test_device and test_feature_point6d

### PR DESCRIPTION
**Pull request by olivier-stasse**
Without clang-format

### CODE REVIEW

**Comment by gabrielebndn** [comment]
Ok, I am checking. So we directly merge to devel, without passing through [topic/fixdevice](https://github.com/stack-of-tasks/sot-core/tree/topic/fixdevice)?

**Comment by gabrielebndn** [comment]
This compiles, but a unit test fails

**Comment by gabrielebndn** [comment]
I think `test_device` does is not working properly. This is the printed output
```
 First integration Sig:Device(simple_humanoid)::output(vector)::state (Type Cst)
simple_humanoid: -0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
sanityCheck: 1
controlInputType:1
simple_humanoid: -0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
-0.5
```
so it seems the state does not change.

File: unitTesting/tools/test_device.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  aDevice.setVelocitySize(38);
+  aDevice.setVelocityBounds(aLowerVelBound, anUpperVelBound);
+  aDevice.setVelocity(aStateVector);
+  aDevice.setState(aStateVector); // entry signal in position
+  aDevice.controlSIN.setConstant(aControlVector);
+
+  for (unsigned int i = 0; i < 2000; i++) {
+    aDevice.stateSOUT.recompute(i);
+    if (i == 1)
+      std::cout << " First integration " << aDevice.stateSOUT << std::endl;
```
Comment (gabrielebndn) [inline unitTesting/tools/test_device.cpp]: This is not the appropriate way of printing the signal value. It just prints
`Sig:Device(simple_humanoid)::output(vector)::state (Type Cst)`

File: unitTesting/tools/test_device.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  aDevice.display(std::cout);
+  aDevice.cmdDisplay();
```
Comment (gabrielebndn) [inline unitTesting/tools/test_device.cpp]: What do these two lines do?

Comment (gabrielebndn) [reply unitTesting/tools/test_device.cpp]: I pushed a fix

File: unitTesting/tools/test_device.cpp (hunk 1 above)
Comment (gabrielebndn) [reply unitTesting/tools/test_device.cpp]: I have pushed a fix

**Comment by jmirabel** [comment]
> It seems the integration is not performed at all. Something is missing.

You may need a setTime(i) on one of the signals (or all).

File: unitTesting/tools/test_device.cpp (hunk 2 above)
Comment (olivier-stasse) [reply unitTesting/tools/test_device.cpp]: Improve coverage

**Comment by gabrielebndn** [comment]
Dear @olivier-stasse: I have recompiled everything making sure there is no confusion concerning the library versions. I put myself in the master Now all unit tests nominally pass, but I still recommend changes. Here are some updates:

- Warnings and crashes on `test_feature_generic` and `test_feature_point_6d` have gone
- I confirm that in `test_feature_point_6d` it seems some `task.taskSOUT` do no coincide with manual. I recommend checking this and putting some automatic checks in place
- I confirm that in `test_device` it seems no integration is performed. This should be fixed, and the value should be checked with `BOOST_CHECK`
-
- Bonus: I had some trouble making the Python installation work. Basically when I tried to execute `from dynamic_graph.sot.core import OpPointModifier` it kept telling me it could not find submodule `sot.core`. After investigating a bit, I discovered `dynamic-graph` is installed in `site-packages` by default, while `sot-core` is installed in `dist-packages`. This is true for the master branch of dynamic-graph-python, as well as for the currently-released v3.3.2 of the same (which is ahead of the master!). So I had to either manually set `PYTHON_STANDARD_LAYOUT` or switch to the devel branch of `dynamic-graph-python` (where the `cmake` submodule has been updated). I tried both, and it worked in both ways. So I recommend a new release of `dynamic-graph-python` including its current `devel` branch.

**Comment by gabrielebndn** [comment]
I have pushed some improvements to test_device: now the result is verified through BOOST_CHECK.
Also, I've pushed an update to the minimum dynamic-graph version

**Pull request merged**